rs/types/wasm_types/src/errors.rs

```diff
@@ -319,6 +355,18 @@ pub enum WasmInstrumentationError {
     InvalidFunctionType(String),
 }
 
+impl MemoryDiskBytes for WasmInstrumentationError {
+    fn heap_bytes(&self) -> usize {
+        match self {
+            WasmInstrumentationError::WasmDeserializeError(wasm_error)
+            | WasmInstrumentationError::WasmSerializeError(wasm_error) => wasm_error.heap_bytes(),
+            WasmInstrumentationError::IncorrectNumberMemorySections { .. } => 0,
+            WasmInstrumentationError::InvalidDataSegment { .. } => 0,
```

Comment:
How about listing the fields explicitly so that new fields result in a compile time error and need to update this implementation?